cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/BaseStorageClient.java

```diff
@@ -0,0 +1,36 @@
+package com.linkedin.openhouse.cluster.storage;
+
+import com.google.common.base.Preconditions;
+import com.linkedin.openhouse.cluster.storage.configs.StorageProperties;
+import org.springframework.util.CollectionUtils;
+
+/**
+ * BaseStorageClient is the interface that holds common functionality for all storage clients like
+ * validating the configuration and storage properties for a given storage type.
+ *
+ * @param <T> native client type for the storage backend.
+ */
+public interface BaseStorageClient<T> {
```

Comment:
can we make the layering synonymous with BaseStorage class? 
ie: `StorageClient -> BaseStorageClient -> HDFSStorageClient`
It may look like:
```
public abstract class BaseStorageClient<T> implements StorageClient<T> {}
public class HdfsStorageClient implements BaseStorageClient<FileSystem> {}
```

That way we can avoid static methods like `static validateProperties`

Reply:
Done